### ISSUE TITLE
Automatic publish plugins

### ISSUE THREAD

**Issue by jsvd** [elastic/logstash#11353]
## State of affairs in Logstash and Plugins

Logstash has hundreds of plugins, each in their own github repository.
Each plugin has a separate release cycle.
All plugins follow semver (major releases break configurations, minors add features, patches bring fixes and documentation improvements).
Plugin documentation is tied to a plugin release + git tag. This means documentation fixes are visible after a plugin (patch) release.
Plugin tests on Travis can take from 3 up to 15 minutes to run.

After reviewing a pull request to a plugin, there should be a new release. 

### Problem: plugins have unreleased stale code in master branch

A plugin release requires:
1. editing the CHANGELOG
2. bumping the plugin version according to the kind of change:
  a) Breaking change: major bump
  b) New features: minor bump
  c) Bug fix or documentation change: patch bump

Not all pull requests bring these two changes. Either the reviewer asks the contributor to do these changes or the reviewer must create a separate commit.

After the pull request is merged CI needs to run and only after it's finished the publish command can be executed successfully.

The reviewer often context switches to something else while waiting for CI to run or if they need to add a separate commit to perform the changelog and version bumps.

This context switch causes code to remain in master branch unreleased.

### Solution: automate publishing

Automatically transition from a reviewed and merged pull request to a new version published in rubygems.

Steps:
1. Detect a new pull request merged and reviewed
2. Wait until destination branch becomes green
  a) If the CI fails trigger notification
3. Detect if changelog+version bump is done
  a) If not, detect the kind of change and push a commit
4. Publish the plugin
5. Add notification to PR about new release containing the change. 

Detecting the change (3.a) will require adding a tag to the commit messages like `[change: minor]` so the script can infer the version bump.

The commit messages will be used as entries to the changelog, so more care is need to have a good commit message before merging, and even some checks (maybe Github Actions?)
